doc/api_reference.md

```diff
@@ -239,6 +240,34 @@ whole will complete with the result of the successor.
 If the predecessor completes with done/error then `func` is not invoked
 and the operation as a whole completes with that done/error signal.
 
+### `let_with_stop_source(Invocable func) -> Sender`
+
+The `let_with_stop_source()` algorithm constructs an
+`inplace_stop_token` that remains alive for the duration of an operation.
+
+`func` is invoked with an lvalue reference to an `inplace_stop_source`
+derived from the `inplace_stop_token`. This invocation must return a
+Sender.
+
+The `inplace_stop_token` is provided to the `Sender` returned by `func`
+via a call to `get_stop_token` on the provided `Receiver`.
+
+The reference passed to `func` remain valid until the returned sender
+completes, at which point the `inplace_stop_token` goes out of scope.
+
+For example:
+```c++
+let_with_stop_source(some_operation(),
+    [](unifex::inplace_stop_source& stop_source) {
+      return other_operation(stop_source);
+    });
+```
+
+If `request_stop` is called on `stop_source` this may cancel some or
+all of `other_operation`. By this mechanism `let_with_stop_source`
+provides the opportunity to cancel work, partially or completely,
+as part of some larger operation.
```

Comment:
The vagueness of not definitely cancelling was what I was going for. You are right that it certainly does request, though.

Reply:
Is the fact that requesting cancellation is not guaranteed to cancel something we assume? Your wording has dropped that.

Presumably it should be assumed generally, though, or we'd have to repeat the text everywhere.

Reply:
Yes, I think some of the wording around whether it might or might not respond to cancellation in a timely manner is intrinsic to the topic of cancellation and probably doesn't need to be spelled out in every algorithm.

From the perspective of this algorithm, all it can do is signal a request to cancel the child operation. It doesn't have any control over what the child operation does in response to this request.